### ISSUE TITLE
Ora2pg not scanning package headers for  global variables and constants (only those declared in the body)

### ISSUE THREAD

**Issue by BuiltByRoberts** [darold/ora2pg#1244]
Hi Gilles

Please note that I am complete newb to Postgres and Ora2pg - Please tell me to RTFM if you like! (but knowing which manual would be useful!)

OK, this is a little weird because while investigating I read this is fixed at a few levels.  I am running 2021 07 02 - v22.1. 

I am checking in global_variables.conf for the globals declared in the Package Header and not finding any of them.

I have tried:
    Removing all comments from around the declarations - No result
    Moving all the global declarations to the body - Appear in global_variables.conf as expected
    I have looked at the config options  - Since under type there is only PACKAGE I don't think there is anything else to add.

SO thinking this is a bug :-(
Merci
Bob

**Comment by edelanoe** [comment]
Hello Gilles, (Happy new year 2022!)

I'm having the same issue: some constants are defined in a package body, together with local types, and nothing is exported.
I've used the master at 2022/01/10. I've tried placing the const before or after some type declarations. No result.
I have tried with a demo package (copy my next comment)

**Comment by edelanoe** [comment]
CREATE OR REPLACE PACKAGE PCK_SAMPLE AS

    PARSE_NoErr             constant number(4) := 0000;     -- Pas d'erreur

    FUNCTION CVS_HEADER RETURN VARCHAR2;


END PCK_SAMPLE;
/
show errors

CREATE OR REPLACE PACKAGE BODY PCK_SAMPLE AS

  FUNCTION CVS_HEADER RETURN VARCHAR2 is
  BEGIN
    return 'Header: /cvsstore/Packages/PCK_SAMPLE.sql,v 1.8 2022/01/10 18:18:36';
  END;

END PCK_SAMPLE;
/
show errors

**Comment by darold** [comment]
You should have a file named global_variables.conf which contain the constant declaration:
```
$ cat global_variables.conf 
# Global variables with default values used in packages.
pck_sample.parse_noerr = '0000'
```
Do you mean that this the case?

**Comment by edelanoe** [comment]
I finally saw it, sorry for the wrong alert. I have global_variables.conf with master dev of 2022/01/10, but not with ora2pg 23.0 (release). Ok for that.

However, package constants should take effect in any call of any function in any session created by any interface program, that is, at server level or at database level. In issue #970, you tell it has to be appended to postgresql.conf. It means to work on the server itself, to do it once per server, and to restart or pg_reload_conf all concerned servers.
This could become a heavy process, for a simple code recompilation. No other ways ? I guess so...

We took the track of defining a schema in which functions return the values of the constants (and the schema comes from an Oracle package). Not always easy to generate, but all the datatypes are permitted.
